### PR TITLE
[vectorized](function) fix array_map funtion return type maybe get wrong

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes
when get function from functionSet, maybe can't change it's any attribute.
but array_map function return type depend on lambda Expr return type,
so new Function object for it.

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnRefExpr.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ protected void analyzeImpl(Analyzer analyzer) throws AnalysisException {
 
     @Override
     protected String toSqlImpl() {
-        return columnName + "(" + columnId + ")";
+        return columnName;
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnRefExpr.java:86]: why change the code ？

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 2]
```diff
@@ -97,14 +99,17 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                 this.setChild(0, lastChild);
             }
 
-            fn = getBuiltinFunction(fnName.getFunction(), argTypes,
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:100]: why not do clone here ？

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 3]
```diff
@@ -97,14 +99,17 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
...
-                    Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
             Expr lambda = this.children.get(0);
+            if (!(lambda instanceof LambdaFunctionExpr)) {
+                throw new AnalysisException("array_map must use lambda as first input params, now is" +
+                        lambda.debugString());
+            }
+            fn = new Function(fnName, Arrays.asList(argTypes), ArrayType.create(lambda.getChild(0).getType(), true),
+                    true, true, NullableMode.DEPEND_ON_ARGUMENT);
             if (fn == null) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:109]: delete the check

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java [hunk 4]
```diff
@@ -199,17 +191,12 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
             }
             fn = getBuiltinFunction(fnName.getFunction(), argTypes,
                     Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
-            if (fn == null) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/LambdaFunctionCallExpr.java:202]: here should not delete ？

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**